### ISSUE TITLE
Windows:  PHP Warning – yii\base\ErrorException unlink(...71.lock): Operation not permitted

### ISSUE THREAD

**Issue by sa3dany** [craftcms/nitro#195]
### Description

Don't know the actual trigger for this but this time I was saving an entry.
```
yii\base\ErrorException: unlink(/home/ubuntu/sites/craft/storage/runtime/mutex/7029da901471d7305219b9d8a659ee71.lock): Operation not permitted in /home/ubuntu/sites/craft/vendor/yiisoft/yii2/mutex/FileMutex.php:161
Stack trace:
#0 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/web/ErrorHandler.php(74): craft\web\ErrorHandler->handleError()
#1 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/mutex/FileMutex.php(161): craft\web\ErrorHandler->handleError()
#2 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/mutex/FileMutex.php(161): ::unlink()
#3 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/mutex/Mutex.php(88): yii\mutex\FileMutex->releaseLock()
#4 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/services/Revisions.php(165): yii\mutex\FileMutex->release()
#5 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/elements/Entry.php(1313): craft\services\Revisions->createRevision()
#6 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/services/Elements.php(2192): craft\elements\Entry->afterPropagate()
#7 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/services/Elements.php(532): craft\services\Elements->_saveElementInternal()
#8 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/controllers/EntriesController.php(376): craft\services\Elements->saveElement()
#9 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): craft\controllers\EntriesController->actionSaveEntry()
#10 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/InlineAction.php:57}()
#11 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#12 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/web/Controller.php(178): craft\controllers\EntriesController->runAction()
#13 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/Module.php(528): craft\controllers\EntriesController->runAction()
#14 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/web/Application.php(291): craft\web\Application->runAction()
#15 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/web/Application.php(559): craft\web\Application->runAction()
#16 /home/ubuntu/sites/craft/vendor/craftcms/cms/src/web/Application.php(270): craft\web\Application->_processActionRequest()
#17 /home/ubuntu/sites/craft/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest()
#18 /home/ubuntu/sites/craft/web/index.php(25): craft\web\Application->run()
#19 {main}
```

### My workaround

This issue also happened to me when I was using a custom Vagrant setup. I solved it by making a change in `config/app.php` and setting and environment variable in `.env`.
```PHP
'components' => [
        'mutex' => function() {
            $config = craft\helpers\App::mutexConfig();
            $config['isWindows'] = (bool)getenv('WINDOWS_FS') ?: null;
            return Craft::createObject($config);
        },
    ],
```

```
WINDOWS_FS=1

# Not sure if this is needed but I feel it is better anyway since these are runtime files
STORAGE_PATH="/var/tmp/<sitename>"
```

### Steps to reproduce

This happened also after I first loaded up nitro, imported a db and then navigated to the home page.

### Additional info

- Nitro version: Just noticed nirto does not have a -v or --version flag (1.0.0-RC1.1)
- Multipass version: 1.3.0

**Comment by jasonmccallister** [comment]
@sa3dany this sounds related to this issue: https://github.com/craftcms/cms/issues/4355#issuecomment-498829513. 

Can you try modifying that components configuration and seeing if it resolves the issue?

**Comment by jasonmccallister** [comment]
@sa3dany I don't know how I missed the entire second half of your issue, but that is the only way we can really detect if the system is using *nix and windows. 

Also, there is a `nitro version` command to see the version, we don't have a flag that shows the version but we could consider adding it.

**Comment by sa3dany** [comment]
@jasonmccallister I've proposed some changes to the readmeto document the fix for this issue in #196.

**Comment by jasonmccallister** [comment]
Merged in the docs, thank you @sa3dany!
